src/main/java/com/github/manolo8/darkbot/gui/components/ApiSettingsPanel.java

```diff
@@ -20,6 +23,49 @@ public ApiSettingsPanel(Config.BotSettings.APIConfig display, JComponent parent)
         super(new MigLayout("wrap 6"));
         this.apiConfig = display;
 
+        if (Main.API.hasCapability(GameAPI.Capability.WINDOW_POSITION)) {
+            JCheckBox c = new JCheckBox();
+            c.setSelected(apiConfig.attachToBot);
+
+            c.addItemListener(l -> apiConfig.attachToBot = l.getStateChange() == ItemEvent.SELECTED);
+
+            add(new JLabel("Attach window: "), "span 3, align left");
+            add(c, "span 3");
+        }
+
+        if (Main.API.hasCapability(GameAPI.Capability.HANDLER_CLEAR_CACHE)) {
+            JButton b = new JButton("Clear Cache");
+            b.addActionListener(l -> Main.API.clearCache());
+            add(b, "span 3, grow");
+        }
+
+        if (Main.API.hasCapability(GameAPI.Capability.HANDLER_CLEAR_RAM)) {
+            JButton b = new JButton("Clear Ram");
+            b.addActionListener(l -> Main.API.emptyWorkingSet());
+            add(b, "span 3, grow");
+        }
```

Comment:
These probably belong in the diagnostics popup menu instead of the api visbility one, but i don't really mind it much